crowdgit/maintainers.py

```diff
@@ -0,0 +1,296 @@
+            AND "maintainerFile" IS NULL
+            AND "lastMaintainerRunAt" < CURRENT_DATE - INTERVAL '30 days'
+        """
+    )
+
+    logger.info(f"Found {len(repos)} repos to reidentify")
+
+    for repo in repos:
+        try:
+            url = repo["url"]
+            repo_id = repo["id"]
+
+            if is_repo_failed(url):
+                logger.info(f"Skipping failed repo: {url}")
+                continue
+
+            parsed_url = urlparse(url)
+            path_parts = parsed_url.path.strip("/").split("/")
+
+            if len(path_parts) >= 2:
+                owner, repo_name = path_parts[:2]
+                if owner == "envoyproxy":
+                    raise Exception("envoy later")
+
+                logger.info(f"Processing repo: github.com/{owner}/{repo_name}")
+                result = scrape(owner, repo_name)
+
+                if result.maintainer_info is not None and result.maintainer_file is not None:
+                    maintainer_file = result.maintainer_file
+                    maintainer_info = result.maintainer_info
+
+                    await process_maintainers(repo_id, url, maintainer_info)
+
+                    # Update githubRepos table
+                    await execute(
+                        """
+                        UPDATE "githubRepos"
+                        SET "maintainerFile" = $1, "lastMaintainerRunAt" = $2
+                        WHERE id = $3
+                    """,
+                        (maintainer_file, datetime.now(), repo_id),
+                    )
+
+                    logger.info(f"Successfully processed maintainers for {owner}/{repo_name}")
+
+                    if GET_COST:
+                        update_cost_csv(url, result.total_cost)
+                else:
+                    await execute(
+                        """
+                        UPDATE "githubRepos"
+                        SET "lastMaintainerRunAt" = $1
+                        WHERE id = $2
+                    """,
+                        (datetime.now(), repo_id),
+                    )
+                    logger.warning(
+                        f"Failed to scrape maintainers for {owner}/{repo_name}: {result.reason or 'Unknown error'}"
+                    )
+
+                    if GET_COST and result.total_cost is not None:
+                        update_cost_csv(url, result.total_cost)
+            else:
+                logger.error(f"Invalid GitHub URL format: {url}")
+                write_failed_repo(url, "", "", "Invalid GitHub URL format")
+
+        except Exception as error:
+            logger.error(f"Error processing repo {url}: {error}")
+            write_failed_repo(url, owner, repo_name, str(error))
+
+
+async def run():
+    async with asyncio.TaskGroup() as tg:
+        tg.create_task(parse_not_parsed())
+        # tg.create_task(parse_already_parsed())
+        # tg.create_task(reidentify_repos_with_no_maintainer_file())
```

Comment:
Also several things here that we do not need.

Reply:
Which ones? Cost related stuff?